### PR TITLE
Add optimization flags for ZCC toolchain

### PR DESCRIPTION
**Pull request by jupiter-zzp**
Build: Change the name of the library used by the ZCC toolchain.
application: Add optimization flags for ZCC toolchain.

### CODE REVIEW

File: Build/toolchain/terapines.mk [hunk 1]
```diff
@@ -16,20 +16,20 @@ endif
 ifneq ($(findstring newlib,$(STDCLIB)),)
 ### Handle cases when STDCLIB variable has newlib in it
 ifeq ($(STDCLIB),newlib_full)
-LDLIBS += -lc_size -lclang_rt.builtins
+LDLIBS += -lc -lclang_rt.builtins
```
Comment (fanghuaqi) [inline Build/toolchain/terapines.mk:19]: Do you have a size version for c library, if yes, you can use c_size version.

**Comment by fanghuaqi** [comment]
Hi @jupiter-zzp , please change this commit into two commits, one commit change the Build folder, another is the application changes

File: Build/toolchain/terapines.mk (hunk 1 above)
Comment (jupiter-zzp) [reply Build/toolchain/terapines.mk:19]: Now ZCC no longer needs to use the _size suffix to search for the size version of the library. ZCC will go to the size subdirectory to find the size version of the library based on the Os flag.

Comment (fanghuaqi) [reply Build/toolchain/terapines.mk:19]: Cool

File: application/baremetal/benchmark/coremark/toolchain_terapines.mk [hunk 2]
```diff
@@ -1 +1 @@
-BENCH_FLAGS ?= -O3 -flto
+BENCH_FLAGS ?= -O3 -flto -falign-functions=4 -falign-loops=4 -flate-loop-unroll -malign-branch -Wl,-mllvm,--align-all-nofallthru-blocks=2 -Wno-unused-command-line-argument
```
Comment (fanghuaqi) [inline application/baremetal/benchmark/coremark/toolchain_terapines.mk:1]: One compiler flags for all core?

**Review by fanghuaqi — CHANGES_REQUESTED** [review]
As describe above, please change one commit to two commit.

- build: update toolchain flags for zcc
- application: update coremark/dhrystone flags for zcc to improve performance

File: application/baremetal/benchmark/coremark/toolchain_terapines.mk (hunk 2 above)
Comment (jupiter-zzp) [reply application/baremetal/benchmark/coremark/toolchain_terapines.mk:1]: Yes

Comment (fanghuaqi) [inline application/baremetal/benchmark/coremark/toolchain_terapines.mk:1]: Can you remove `-Wno-unused-command-line-argument`  and remove the unused options not supported by Zcc.

File: application/baremetal/benchmark/dhrystone/toolchain_terapines.mk [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ifeq ($(DHRY_MODE),ground)
-BENCH_FLAGS ?= -O3 -flto -fno-inline -fno-builtin-printf -funroll-loops -falign-functions=4 -falign-loops=4
+BENCH_FLAGS ?= -Ofast -flto -fno-inline -fno-builtin-printf -funroll-loops -falign-functions=4 -falign-loops=4 -flate-loop-unroll -malign-branch -Wl,-mllvm,--align-all-nofallthru-blocks=2 -Wno-unused-command-line-argument
```
Comment (fanghuaqi) [inline application/baremetal/benchmark/dhrystone/toolchain_terapines.mk:2]: Same as above

**Review by fanghuaqi — CHANGES_REQUESTED** [review]
Please check and update it.

File: application/baremetal/benchmark/coremark/toolchain_terapines.mk (hunk 2 above)
Comment (jupiter-zzp) [reply application/baremetal/benchmark/coremark/toolchain_terapines.mk:1]: "-Wl,-mllvm,--align-all-nofallthru-blocks=2" is only used in the linking phase and will raise a warning in the compilation phase. "-Wno-unused-command-line-argument" is used to dismiss this warning.
Or I can use 
```
BENCH_FLAGS ?= -O3 -flto -falign-functions=4 -falign-loops=4 -flate-loop-unroll -malign-branch
LDFLAGS += -Wl,-mllvm,--align-all-nofallthru-blocks=2
```
to replace it?

**Pull request merged**